spock-core/src/main/java/spock/lang/Snapshotter.java

```diff
@@ -276,7 +277,7 @@ public Store(IterationInfo iterationInfo, Path rootPath, boolean updateSnapshots
       this.extension = Assert.notNull(extension);
       this.charset = Assert.notNull(charset);
 
-      Class<?> specClass = iterationInfo.getFeature().getSpec().getReflection();
+      Class<?> specClass = iterationInfo.getFeature().getParent().getBottomSpec().getReflection();
```

Comment:
💬 Allows defining snapshot-based in base classes, but child specs can overwrite snapshots.

Reply:
I added release notes in [`49d2685` (#2112)](https://github.com/spockframework/spock/pull/2112/commits/49d2685e9a5b4fde8d201bda3ce1e846ad922a1c). I think there is value in supporting both scenarios: you may want to define features and snapshots for base class, but overwrite them for certain features.